ui/pages/confirmations/components/confirm/splash/smart-account-update/smart-account-update.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useState } from 'react';
+import { NameType } from '@metamask/name-controller';
+import { TransactionMeta } from '@metamask/transaction-controller';
+
+import { SMART_ACCOUNT_INFO_LINK } from '../../../../../../../shared/lib/ui-utils';
+import {
+  Box,
+  Button,
+  ButtonSize,
+  ButtonVariant,
+  Text,
+} from '../../../../../../components/component-library';
+import {
+  AlignItems,
+  BackgroundColor,
+  BlockSize,
+  BorderRadius,
+  Display,
+  FlexDirection,
+  FontWeight,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../../hooks/useI18nContext';
+import Name from '../../../../../../components/app/name';
+import { useConfirmContext } from '../../../../context/confirm';
+import { useSmartAccountActions } from '../../../../hooks/useSmartAccountActions';
+
+export function SmartAccountUpdate() {
+  const [acknowledged, setAcknowledged] = useState(false);
+
+  const t = useI18nContext();
+  const { currentConfirmation } = useConfirmContext<TransactionMeta>();
+  const { chainId, txParams } = currentConfirmation ?? {};
+  const { from } = txParams;
+
+  const { handleRejectUpgrade } = useSmartAccountActions();
+
+  if (!currentConfirmation || acknowledged) {
+    return null;
+  }
+
+  return (
+    <Box
```

Comment:
do we not have a reusable component for the modals? I think we should check with design on expected responsive behavior. Also, when the view height is shrunken, the modal is not scrollable 

![CleanShot 2025-04-09 at 11 23 22](https://github.com/user-attachments/assets/9e6f163a-abc9-4c1d-bf17-73a152cf3d19)

https://github.com/user-attachments/assets/98679805-a1e9-432c-a559-9b0aa7606380

Reply:
hm. for some reason this is what appears opening MM in full screen 

![CleanShot 2025-04-09 at 12 06 00](https://github.com/user-attachments/assets/64a4867c-dc3a-4b1d-a8f8-09285384a4fd)

will need to fix these buttons

Reply:
It changes based on the height

https://github.com/user-attachments/assets/3da89ab8-af32-4c46-869b-0886c4fe9cf0

Reply:
Agree that this isn't intended to be a modal, but a full page view until the user selects an option.

Reply:
I see. Is it still possible to update the style to be more inline with other UI/UX? I'm wondering about:
- buttons should maintain height
- buttons should have consistent space between when stacked, I think similar to figma design
- list of description elements could be right aligned
- potential to add max width on image 

Will need to be aware in case other splash ui/ux comes up that we may want to make reusable components